libs/libcommon/src/libcommon/parquet_utils.py

```diff
@@ -601,31 +601,3 @@ def query_truncated_binary(self, offset: int, length: int) -> tuple[pa.Table, li
             f" split={self.split}, offset={offset}, length={length}, with truncated binary"
         )
         return self.parquet_index.query_truncated_binary(offset=offset, length=length)
-
-
-class Indexer:
-    def __init__(
-        self,
-        parquet_metadata_directory: StrPath,
-        httpfs: HTTPFileSystem,
-        max_arrow_data_in_memory: int,
-    ):
-        self.parquet_metadata_directory = parquet_metadata_directory
-        self.httpfs = httpfs
-        self.max_arrow_data_in_memory = max_arrow_data_in_memory
-
-    @lru_cache(maxsize=1)
```

Comment:
The only breaking change is that `lru_cache` is not applied when constructing the index. I assume this caching didn't have much of an effect on runtime performance, @lhoestq please confirm.